inspirehep/modules/workflows/tasks/classifier.py

```diff
@@ -76,33 +76,48 @@ def _classify_paper(obj, eng):
             extract_acronyms=extract_acronyms
         )
 
-        fulltext_used = True
+        try:
+            data = get_value(obj.data, 'titles.title', [])
+            data.extend(get_value(obj.data, 'titles.subtitle', []))
+            data.extend(get_value(obj.data, 'abstracts.value', []))
+            data.extend(get_value(obj.data, 'keywords.value', []))
+            if not data:
+                obj.log.error("No classification done due to missing data.")
+                data_result = None
+            else:
+                data_result = get_keywords_from_text(data, **params)
+                if data_result:
+                    data_result['complete_output'] = clean_instances_from_data(
+                        data_result.get("complete_output", {})
+                    )
+        except ClassifierException as e:
+            obj.log.exception(e)
+            data_result = None
+
         with get_document_in_workflow(obj) as tmp_document:
             try:
                 if tmp_document:
                     result = get_keywords_from_local_file(tmp_document, **params)
+                    fulltext_used = True
+                    if result:
+                        result['complete_output'] = clean_instances_from_data(
+                            result.get("complete_output", {})
+                        )
                 else:
-                    data = get_value(obj.data, 'titles.title', [])
-                    data.extend(get_value(obj.data, 'titles.subtitle', []))
-                    data.extend(get_value(obj.data, 'abstracts.value', []))
-                    data.extend(get_value(obj.data, 'keywords.value', []))
-                    if not data:
-                        obj.log.error("No classification done due to missing data.")
-                        return
-                    result = get_keywords_from_text(data, **params)
+                    result = data_result
                     fulltext_used = False
             except ClassifierException as e:
                 obj.log.exception(e)
                 return
```

Comment:
You probably shouldn't `return` here, as you still want to get the keywords from metadata into the record in this case.

Reply:
:+1: